### PR TITLE
Update the way default contribution amount is handled.

### PR DESCRIPTION
**Pull request by bsclifton**
Instead of having a set value, the current value is read from a config.
Value is then locked in once a user chooses a value OR has a non-zero balance (ex: either a grant or adding funds)

Fixes https://github.com/brave/browser-laptop/issues/12573

Auditors: @mrose17

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [x] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

## Test Plan:
See https://github.com/brave/browser-laptop/issues/12573

## Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

**Comment by bsclifton** [comment]
Security review submitted here:
https://github.com/brave/internal/issues/183

File: js/constants/appConfig.js [hunk 1]
```diff
@@ -101,7 +101,8 @@ module.exports = {
     url: adHost
   },
   payments: {
-    delayNotificationTryPayments: 1000 * 60 * 60 * 24 * 10 // 10 days (from firstRunTimestamp)
+    delayNotificationTryPayments: 1000 * 60 * 60 * 24 * 10, // 10 days (from firstRunTimestamp)
+    defaultContributionAmount: 5
```
Comment (bsclifton) [inline js/constants/appConfig.js]: Here's where the new default is set as 5 BAT

File: js/constants/appConfig.js [hunk 2]
```diff
@@ -187,7 +188,7 @@ module.exports = {
     // payments
     'payments.allow-media-publishers': true,
     'payments.allow-non-verified-publishers': true,
-    'payments.contribution-amount': 10, // BAT
+    'payments.contribution-amount': null, // BAT
```
Comment (bsclifton) [inline js/constants/appConfig.js]: The actual value (for new installs only) is set to null to avoid "locking in" a value

**Comment by LaurenWags** [comment]
Test Cases 1&2 were as expected.
Test Case 3 (https://github.com/brave/browser-laptop/issues/12573) was not. In step 8 after relaunching Brave, my monthly budget was 10, not 5.
Additional test case performed (since staging still had test grants), similar to TC 3 but funded wallet with test grant instead of funds from uphold sandbox. In this additional test, the monthly budget remained 5, did not change to 10.

File: app/browser/api/ledger.js [hunk 3]
```diff
@@ -1562,6 +1575,7 @@ const onWalletProperties = (state, body) => {
   const balance = parseFloat(body.get('balance'))
   if (balance >= 0) {
     state = ledgerState.setInfoProp(state, 'balance', balance)
+    lockInContributionAmount()
```
Comment (mrose17) [inline app/browser/api/ledger.js]: shouldn't `balance` be passed as the parameter?

Comment (bsclifton) [reply app/browser/api/ledger.js]: You are absolutely right! doh (great catch!)

**Comment by bsclifton** [comment]
@LaurenWags the test cases you tried which failed were because of the item @mrose17 caught above 😄  I manually tested after updating and verified it works as expected

**Comment by bsclifton** [comment]
master 43ba3f0658ca4fa748772a240b66d5b5f84b775c
0.21.x eba07479ed4ad2d0f3c13735cafe25c85d6b465a
0.20.x 8b002f073e08215e8f0a22929a068b258dc1911e
0.19.x 37ec77441df3c03672b52dbbf1d94a65f38eddd2

**Pull request merged**